src/Simplex/Messaging/Agent/Transmission.hs

```diff
@@ -91,13 +90,50 @@ data ACommand (p :: AParty) where
 
 deriving instance Show (ACommand p)
 
+type Message = ByteString
+
 data AMessage where
   HELLO :: VerificationKey -> AckMode -> AMessage
   REPLY :: SMPQueueInfo -> AMessage
   A_MSG :: MsgBody -> AMessage
 
--- A_ACK :: AgentMsgId -> AckStatus -> AMessage
--- A_DEL :: AMessage
+parseMessage :: Message -> Either ErrorType AMessage
+parseMessage msg = case B.words msg of
+  -- ["HELLO", key, ackMode] -> Right $ HELLO key $ ackMode am
+  ["REPLY", qInfo] -> REPLY <$> parseSmpQueueInfo qInfo
+  ["A_MSG", msgBody] -> Right $ A_MSG msgBody
```

Comment:
That probably needs to change using the same approach as in https://github.com/simplex-chat/simplex-messaging/blob/master/src/Simplex/Messaging/Agent/Transmission.hs#L303

But not sure about it, I didn't think about variable size bodies... Let's think if there is a better approach to storing messages in the database other than just ByteString... Maybe we can somehow store fields. Not sure. It should not hold merging.

Reply:
This question depends on whether we want to use the same parsing/serialization here as in the protocol (I thought we would use the same) where the size needs to be used in the syntax so that the end of the message can be known with binary that follows and can be padded.